airflow/models/taskinstance.py

```diff
@@ -3244,6 +3244,7 @@ def fetch_handle_failure_context(
     @provide_session
     def save_to_db(ti: TaskInstance | TaskInstancePydantic, session: Session = NEW_SESSION):
         ti = _coalesce_to_orm_ti(ti=ti, session=session)
+        ti.updated_at = timezone.utcnow()
```

Comment:
In that case, don't we want to simply align the value of `updated_at` with the value of `end_date`? (i.e, `ti.updated_at = ti.end_date`) 
What is the purpose of `utcnow()` as the value of `updated_at`? It will be a just date in a delay of some miliseconds after the `end_date` - what is special about this value?

Reply:
I can confirm the same issue happens even if the task is not SKIPPED or SUCCESS. 
Here is an example result when testing with a task that only raises AirflowTaskTimeout, showing `updated_at` is smaller than `end_date`. 
<img width="1126" alt="Screenshot 2024-07-17 at 12 19 35 AM" src="https://github.com/user-attachments/assets/79deeffb-c623-4d4b-82f3-97e97b68fed9">

Because `save_to_db` is used with all possible return state, I believe it's a better and easier place fixing this issue.

Reply:
Firstly I think `save_to_db` is the best place to fix this issue to make sure all db writes have updated `updated_at`. Also, `save_to_db` is not only called when task ended, but also before the starting of the task and failed due to timeout or other reasons (see `_run_raw_task`), and end_date could be empty in some of those cases.

Reply:
Thanks for the clarification! It makes sense.
LGTM then.